emcc.py

```diff
@@ -4201,12 +4201,9 @@ def generate_traditional_runtime_html(target, options, js_target, target_basenam
 
   html_contents = do_replace(shell, '{{{ SCRIPT }}}', script.replacement())
   html_contents = tools.line_endings.convert_line_endings(html_contents, '\n', options.output_eol)
-
-  try:
-    # Force UTF-8 output for consistency across platforms and with the web.
-    utils.write_binary(target, html_contents.encode('utf-8'))
-  except OSError as e:
-    exit_with_error(f'cannot write output file: {e}')
+  if os.path.isdir(target):
+    exit_with_error(f'cannot write output file `{target}`: Is a directory')
```

Comment:
Is there a reason to handle the "is a directory" error differently than any other error?

Reply:
Just for consistency with how we do for the main `.js` and `.wasm` output files already.   

This way we don't see a python backtrace in that case.   Perhaps we should have a helper called write_file_or_die() that handles this for us, but for now I'm just mimicing the code in `move_file` which is used to move `.js` and `.wasm` files into their final locations.

Reply:
You can see how the test now expects that exact same failure mode for the `.js` was the `.html` whereas before they were different.

Reply:
I extracted a function to make this more clear